RecoEgamma/PhotonIdentification/plugins/PhotonMVANtuplizer.cc

```diff
@@ -0,0 +1,370 @@
+       tree_->Branch(mvaCatBranchNames_[k].c_str() ,  &mvaCats_[k]);
+   }
+}
+
+
+PhotonMVANtuplizer::~PhotonMVANtuplizer()
+{
+
+   // do anything here that needs to be done at desctruction time
+   // (e.g. close files, deallocate resources etc.)
+
+}
+
+
+//
+// member functions
+//
+
+// ------------ method called for each event  ------------
+void
+PhotonMVANtuplizer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+    // Fill global event info
+    nEvent_ = iEvent.id().event();
+    nRun_   = iEvent.id().run();
+    nLumi_  = iEvent.luminosityBlock();
+
+
+    // Retrieve Vertecies
+    edm::Handle<reco::VertexCollection> vertices;
+    iEvent.getByToken(vertices_, vertices);
+    if( !vertices.isValid() ){
+      iEvent.getByToken(verticesMiniAOD_,vertices);
+      if( !vertices.isValid() )
+        throw cms::Exception(" Collection not found: ")
+          << " failed to find a standard AOD or miniAOD vertex collection " << std::endl;
+    }
+
+    vtxN_ = vertices->size();
+
+    // Retrieve Pileup Info
+    edm::Handle<std::vector< PileupSummaryInfo > >  pileup;
+    iEvent.getByToken(pileup_, pileup);
+    if( !pileup.isValid() ){
+      iEvent.getByToken(pileupMiniAOD_,pileup);
+      if( !pileup.isValid() )
+        throw cms::Exception(" Collection not found: ")
+          << " failed to find a standard AOD or miniAOD pileup collection " << std::endl;
+    }
+
+    // Fill with true number of pileup
+    if(true) {
+       for(const auto& pu : *pileup)
+       {
+           int bx = pu.getBunchCrossing();
+           if(bx == 0)
+           {
+               genNpu_ = pu.getPU_NumInteractions();
+               break;
+           }
+       }
+    }
+
+
+    edm::Handle<edm::View<reco::Photon> > src;
+
+    // Retrieve the collection of particles from the event.
+    // If we fail to retrieve the collection with the standard AOD
+    // name, we next look for the one with the stndard miniAOD name.
+    iEvent.getByToken(src_, src);
+    if( !src.isValid() ){
+      iEvent.getByToken(srcMiniAOD_,src);
+      if( !src.isValid() )
+        throw cms::Exception(" Collection not found: ")
+          << " failed to find a standard AOD or miniAOD particle collection " << std::endl;
+    }
+
+    // Get MVA decisions
+    edm::Handle<edm::ValueMap<bool> > decisions[nEleMaps_];
+    for (size_t k = 0; k < nEleMaps_; ++k) {
+        iEvent.getByToken(phoMapTokens_[k],decisions[k]);
+    }
+
+    // Get MVA values
+    edm::Handle<edm::ValueMap<float> > values[nValMaps_];
+    for (size_t k = 0; k < nValMaps_; ++k) {
+        iEvent.getByToken(valMapTokens_[k],values[k]);
+    }
+
+    // Get MVA categories
+    edm::Handle<edm::ValueMap<int> > mvaCats[nCats_];
+    for (size_t k = 0; k < nCats_; ++k) {
+        iEvent.getByToken(mvaCatTokens_[k],mvaCats[k]);
+    }
+
+    int nEle = src->size();
+
+    for(int iPho = 0; iPho < nEle; ++iPho) {
+
+        const auto pho =  src->ptrAt(iPho);
+
+        if (pho->pt() < 5) {
```

Comment:
Where is this "5" taken from?
Shouldn't it be better make it configurable? If it is carved in stone, maybe a comment here can be added